src/OrchardCore.Modules/OrchardCore.Contents/Indexing/FullTextContentIndexHandler.cs

```diff
@@ -21,10 +22,13 @@ public async Task BuildIndexAsync(BuildIndexContext context)
             // Index each segment as a new value to prevent from allocation a new string
             foreach (var segment in result.Segments)
             {
-                context.DocumentIndex.Set(
-                    IndexingConstants.FullTextKey,
-                    segment,
-                    DocumentIndexOptions.Analyze | DocumentIndexOptions.Sanitize);
+                if (!String.IsNullOrEmpty(segment))
```

Comment:
I added this check, because by default Widgets often don't have display text, or a body aspect, so many of the segments were returning empty, and it seemed worthwhile to not place empty strings into the index.